lambdas/issuecredential/src/test/java/uk/gov/di/ipv/cri/kbv/api/service/VerifiableCredentialServiceTest.java

```diff
@@ -94,27 +96,32 @@ void shouldReturnASignedVerifiableCredentialJwt(
             int answeredInCorrectly,
             int expectedVerificationScore,
             ContraIndicator expectedContraIndicator)
-            throws JOSEException, InvalidKeySpecException, NoSuchAlgorithmException,
-                    ParseException {
+            throws JOSEException, InvalidKeySpecException, NoSuchAlgorithmException, ParseException,
+                    JsonProcessingException {
         initMockConfigurationService();
         SignedJWTFactory signedJwtFactory = new SignedJWTFactory(new ECDSASigner(getPrivateKey()));
+        spyEvidenceFactory =
+                spy(
```

Comment:
Is there any specific for reason a `spy` here?
Given that the `EvidenceFactory` has it's own unit tests then a `Mock` could be used instead. Ultimately the unit under test is the `VerifiableCredential` class & we are not concerned with the inner workings of the `EvidenceFactory` dependency

Reply:
@tal-nagra it is b'cos I have to forward the same instance `ObjectMapper` used by VerifiableCredentialService  to the constructor of `EvidenceFactory`  so a Mock does work here

Reply:
@tal-nagra The reason is b'cos the `ObjectMapper` is not a Mock, and it is passed on to the constructor of the EvidenceFactory, if ObjectMapper was a mock then I maybe fine. But I think the test would be longer as I have to mock the call to ObjectMapper.  It would take more work to get this to really work